### ISSUE TITLE
Try out Deepseek

### ISSUE THREAD

**Issue by kongzii** [gnosis/prediction-market-agent#557]
https://api-docs.deepseek.com/news/news1120

:mag: o1-preview-level performance on AIME & MATH benchmarks.
:bulb: Transparent thought process in real-time.
:hammer_and_wrench: Open-source models & API coming soon!

**Comment by empeje** [comment]
Thanks @gabrielfior for the reply, I'm going to experiment with it.

It's nice that test and benchmark are available.

**Comment by kongzii** [comment]
We do have a benchmark to compare agent implementations as well https://github.com/gnosis/prediction-market-agent/blob/main/prediction_market_agent/agents/think_thoroughly_agent/benchmark.py#L101. But of course it's not as accurate as just running the agents for some time and then checking it backwards after markets are resolved.

**Comment by empeje** [comment]
Update to my research, it looks like we can directly test with DeepSeek by just changing the OpenAI URL and pass the token of DeepSeek there. See the following code from their official documentation

```python
# pip3 install langchain_openai
# python3 deepseek_v2_langchain.py
from langchain_openai import ChatOpenAI

llm = ChatOpenAI(
    model='deepseek-chat', 
    openai_api_key='<your api key>', 
    openai_api_base='https://api.deepseek.com',
    max_tokens=1024
)

response = llm.invoke("Hi!")
print(response.content)
```

**Comment by empeje** [comment]
If I were to implement the DeepSeek, what is the common way in this current codebase to do it? I can imagine I can just change the model type, but wondering if we should have a feature flag to switch between this agent so we can compare models.

**Comment by kongzii** [comment]
> a feature flag to switch between this agent so we can compare models.

Yes, that's good thinking. Good agent to start is Prophet family, [check out this file.](https://github.com/gnosis/prediction-market-agent/blob/main/prediction_market_agent/agents/prophet_agent/deploy.py#L165).

There, you will see that each agent is a separate class, and each class comes with its own `model` configuration:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/121f85b9-eb86-415f-9dfa-9900cc27e187)

This was fine until now as all Prophets are using openai, however now you will need to also change the API BASE URL, not just the model string, as you shown before. So that's something you need to add in there.

The Prophet agent itself is defined [in another repository ](https://github.com/agentcoinorg/predictionprophet), so you may need to do some modifications there as well.

**Comment by kongzii** [comment]
@empeje Do you still want to work on this? If not, I'll do it as part of https://github.com/gnosis/prediction-market-agent/issues/683

**Comment by kongzii** [comment]
Done
